tests/test_backend.py

```diff
@@ -0,0 +1,190 @@
+
+        response = await scrapy_backend(req)
+
+        mock_downloader.assert_called_once()
+        assert isinstance(response, web_poet.HttpResponse)
+
+        assert response.url == "https://example.com"
+        assert response.text == "some content"
+        assert response.status == 200
+        assert response.headers.get("Content-Type") == "text/html; charset=utf-8"
+        assert len(response.headers) == 1
+
+
+@ensureDeferred
+async def test_scrapy_poet_backend_ignored_request():
+    """It should handle IgnoreRequest from Scrapy according to the web poet
+    standard on additional request error handling."""
+    req = web_poet.HttpRequest("https://example.com")
+
+    with mock.patch(
+        "scrapy_poet.backend.maybe_deferred_to_future", new_callable=AsyncMock
+    ) as mock_dtf:
+        mock_dtf.side_effect = scrapy.exceptions.IgnoreRequest
+        mock_downloader = mock.MagicMock(return_value=AsyncMock)
+        scrapy_backend = create_scrapy_backend(mock_downloader)
+
+        with pytest.raises(web_poet.exceptions.HttpError):
+            await scrapy_backend(req)
+
+
+@ensureDeferred
+async def test_scrapy_poet_backend_twisted_error():
+    req = web_poet.HttpRequest("https://example.com")
+
+    with mock.patch(
+        "scrapy_poet.backend.maybe_deferred_to_future", new_callable=AsyncMock
+    ) as mock_dtf:
+        mock_dtf.side_effect = twisted.internet.error.TimeoutError
+        mock_downloader = mock.MagicMock(return_value=AsyncMock)
+        scrapy_backend = create_scrapy_backend(mock_downloader)
+
+        with pytest.raises(web_poet.exceptions.HttpRequestError):
+            await scrapy_backend(req)
+
+
+@ensureDeferred
+async def test_scrapy_poet_backend_head_redirect(fake_http_response):
+    req = web_poet.HttpRequest("https://example.com", method="HEAD")
+
+    with mock.patch(
+        "scrapy_poet.backend.maybe_deferred_to_future", new_callable=AsyncMock
+    ) as mock_dtf:
+        mock_dtf.return_value = fake_http_response
+        mock_downloader = mock.MagicMock(return_value=AsyncMock)
+        scrapy_backend = create_scrapy_backend(mock_downloader)
+
+        await scrapy_backend(req)
+
+        args, kwargs = mock_downloader.call_args
+        scrapy_request = args[0]
+        assert scrapy_request.meta.get("dont_redirect") is True
+
+
+@ensureDeferred
+async def test_scrapy_poet_backend_dont_filter(fake_http_response):
+    req = web_poet.HttpRequest("https://example.com")
+
+    with mock.patch(
+        "scrapy_poet.backend.maybe_deferred_to_future", new_callable=AsyncMock
+    ) as mock_dtf:
+        mock_dtf.return_value = fake_http_response
+        mock_downloader = mock.MagicMock(return_value=AsyncMock)
+        scrapy_backend = create_scrapy_backend(mock_downloader)
+
+        await scrapy_backend(req)
+
+        args, kwargs = mock_downloader.call_args
+        scrapy_request = args[0]
+        assert scrapy_request.dont_filter is True
```

Comment:
I wonder how hard would it be to have integration-style tests for such features, similar to https://github.com/scrapy-plugins/scrapy-splash/blob/f5273b3a0ef15a9f6809d305a1dbe8f3efd12c5f/tests/test_integration.py#L113. My worry is that Scrapy has many components, which interact in various ways, so it's possible we miss something if we rely on mock-style tests. What do you think, would it worth it?